### PR TITLE
Tests for 3rdPartyRewards and nativeRewards

### PR DESCRIPTION
**Pull request by AlexChetverov**

### CODE REVIEW

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+        expect(res.data).toEqual("NotAPromotedPool");
+      });
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:143]: this test is about claiming too, plz ensure that claiming fails too.

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:159]: what about the second part of the test? 

```
waits and claims all rewards THEN operation is successful
```

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+        await ProofOfStake.claim3rdpartyRewards(liqId, newToken),
+        testUser.keyRingPair,
+      );
+
+      await testUser.refreshAmounts(AssetWallet.AFTER);
+      const rewardTokenAfter =
+        await testUser.getAsset(newToken)?.amountAfter.free!;
+
+      expect(rewardTokenBefore).bnEqual(BN_ZERO);
+      expect(rewardTokenAfter).bnGt(BN_ZERO);
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:212]: :clap:

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+      testUser.addAsset(newToken3);
+      await testUser.refreshAmounts(AssetWallet.BEFORE);
+      const rewardToken1Before =
+        await testUser.getAsset(newToken)?.amountBefore.free!;
+      const rewardToken3Before =
+        await testUser.getAsset(newToken3)?.amountBefore.free!;
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.claim3rdpartyRewards(liqId, newToken),
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:286]: should not we use the claimAll extrinsic?  I think that is what the test is about, ensure that claimAll claim both token1 and token3 
maybe Im missign sth :thinking:

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+  });
+});
+
+async function createAndPromoteSecondPool(
+  userLiqTokenAmount: BN,
+  isRewardPoolEnabled: boolean,
+) {
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintToken(newToken, sudo, Assets.DEFAULT_AMOUNT.muln(40e6)),
+    Assets.mintToken(liqId, testUser, userLiqTokenAmount),
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:348]: plz do not use global variables in a local scope. send them as a parameter in the function.

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        Assets.DEFAULT_AMOUNT.muln(20e6),
+        newToken,
+        Assets.DEFAULT_AMOUNT.muln(20e6),
+      ),
+    ),
+    Assets.promotePool(liqId.toNumber(), 20),
+  );
+  liqId2 = await getLiquidityAssetId(MGA_ASSET_ID, newToken);
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:360]: do not modify global variables inside a function. Plz return it and modify it in the test.
There is no need to keep that variable globally. It must be a test coped attribute. 
Plz remove it globally and return it to that function, so it's available in the test scope.

File: test/parallel/proofOfStake.nativeRewards.test.ts [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    Assets.promotePool(liqId.toNumber(), 20),
+    Assets.mintToken(liqId, testUser, Assets.DEFAULT_AMOUNT),
+  );
+
+  testUser.addAsset(liqId);
+  testUser.addAsset(MGA_ASSET_ID);
+  testUser.addAsset(token1);
+});
+
+test("activate some Liquidity using SDK THEN claim rewards THEN deactivate Liquidity", async () => {
```
Comment (goncer) [inline test/parallel/proofOfStake.nativeRewards.test.ts:86]: quite lost here, where is the SDK used?

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+        sudo,
+        Xyk.createPool(
+          MGA_ASSET_ID,
+          Assets.DEFAULT_AMOUNT.muln(20e6),
+          newToken2,
+          Assets.DEFAULT_AMOUNT.muln(20e6),
+        ),
+      ),
+    );
+    liqId = await getLiquidityAssetId(MGA_ASSET_ID, newToken2);
```
Comment (goncer) [inline test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:70]: can you please rename it by: 
LiqIdMgaToken2? 
ITs quite difficult to understand the test when there are more than one liq token :)

**Review by goncer — CHANGES_REQUESTED** [review]

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts (hunk 4 above)
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:286]: Function `proofOfStake.claimRewardsAll` doesn't work with 3rd party rewards, so we can't use this one

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts (hunk 8 above)
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:70]: Renamed all liqId tokens

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts (hunk 1 above)
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:143]: Done

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts (hunk 2 above)
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:159]: Done

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts (hunk 5 above)
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:348]: added var in function

File: test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts (hunk 6 above)
Comment (AlexChetverov) [reply test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts:360]: added var in function

**Review by goncer — APPROVED** [review]

**Review by goncer — APPROVED** [review]

**Pull request merged**